config/routes.rb

```diff
@@ -40,4 +42,10 @@
     end
     resources :principals, only: [:index, :show]
   end
+
+  Sidekiq::Web.use Rack::Auth::Basic do |username, password|
+    Authentication.authenticate(username, password)
+  end if Rails.env.production?
```

Comment:
As discussed, stylistically my preference would be to wrap the whole block in the if statement.